sqlmesh/core/renderer.py

```diff
@@ -206,19 +209,20 @@ def _resolve_tables(
 
         expression = expression.copy()
         with _normalize_and_quote(expression, self._dialect) as expression:
-            snapshots = snapshots or {}
+            snapshots = snapshots or []
             table_mapping = table_mapping or {}
-            mapping = {**to_table_mapping(snapshots.values(), deployability_index), **table_mapping}
-            expand = {d.normalize_model_name(name, dialect=self._dialect) for name in expand} | {
-                name for name, snapshot in snapshots.items() if snapshot.is_embedded
-            }
+            mapping = {**to_table_mapping(snapshots, deployability_index), **table_mapping}
+            expand = set(expand) | {snapshot.fqn for snapshot in snapshots if snapshot.is_embedded}
```

Comment:
bring back the normalizer here

Reply:
anywhere that passes in strings, it should be considered needing normalization and expand is strings

Reply:
We discussed and this suggestion breaks the principle of normalizing at the edge and therefore we are leaving as is